android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/FxaAccountManager.kt

```diff
@@ -528,15 +527,13 @@ open class FxaAccountManager(
         via: Event,
     ): Event? = when (forState.progressState) {
         ProgressState.Initializing -> {
-            when (accountOnDisk) {
```

Comment:
I think it's a good idea, but I'm happy to split that out from the rest of the changes though if you have concerns.

The main reason is that `getAuthState()` should be more accurate.  From what I can tell, just because an account is stored on disk doesn't mean the account is actually connected.  It could also be in the `AuthenticationProblems` state.  Also `getAuthState()` is defined in the app-services component so this simplifies moving the code over.

Reply:
It might be actually a good idea, it's just something that caught my eye, cause it's not just a refactoring, but might affect the business logic. I am fine with a change, but I need more explanation/context to be sure that we aren't breaking anything. Given that it's a business logic change, a separate PR focused on it might be better.